R/derive_vars_computed.R

```diff
@@ -0,0 +1,211 @@
+#'
+#'   *Permitted Values:* List of variable-value pairs
+#'
+#' @param keep_nas Keep observations with `NA`s
+#'
+#'   If the argument is set to `TRUE`, observations are added even if some of
+#'   the values contributing to the computed value are `NA`.
+#'
+#' @details For each group (with respect to the variables specified for the
+#'   `by_vars` parameter), the value of the new variable is computed in the
+#'   output dataset if the filtered input dataset (`dataset`) or the additional
+#'   dataset (`dataset_add`) contains exactly one observation for each parameter
+#'   code specified for `parameters`.
+#'
+#' @return The input dataset with the new variable added.
+#'
+#' @family der_prm_bds_findings
+#'
+#' @keywords der_prm_bds_findings
+#'
+#' @export
+#'
+#' @examples
+#' library(tibble)
+#' library(dplyr)
+#'
+#' # Example 1: Derive BMIBL
+#' adsl <- tribble(
+#'   ~STUDYID,   ~USUBJID, ~AGE,   ~AGEU,
+#'   "PILOT01", "01-1302",   61, "YEARS",
+#'   "PILOT01", "17-1344",   64, "YEARS"
+#' )
+#'
+#' advs <- tribble(
+#'   ~STUDYID, ~USUBJID, ~PARAMCD, ~PARAM, ~VISIT, ~AVAL, ~AVALU, ~ABLFL,
+#'   "PILOT01", "01-1302", "HEIGHT", "Height (cm)", "SCREENING", 177.8, "cm", "Y",
+#'   "PILOT01", "01-1302", "WEIGHT", "Weight (kg)", "SCREENING", 81.19, "kg", "N",
+#'   "PILOT01", "01-1302", "WEIGHT", "Weight (kg)", "BASELINE", 82.1, "kg", "Y",
+#'   "PILOT01", "01-1302", "WEIGHT", "Weight (kg)", "WEEK 2", 81.19, "kg", "N",
+#'   "PILOT01", "01-1302", "WEIGHT", "Weight (kg)", "WEEK 4", 82.56, "kg", "N",
+#'   "PILOT01", "01-1302", "WEIGHT", "Weight (kg)", "WEEK 6", 80.74, "kg", "N",
+#'   "PILOT01", "17-1344", "HEIGHT", "Height (cm)", "SCREENING", 163.5, "cm", "Y",
+#'   "PILOT01", "17-1344", "WEIGHT", "Weight (kg)", "SCREENING", 58.06, "kg", "N",
+#'   "PILOT01", "17-1344", "WEIGHT", "Weight (kg)", "BASELINE", 58.06, "kg", "Y",
+#'   "PILOT01", "17-1344", "WEIGHT", "Weight (kg)", "WEEK 2", 58.97, "kg", "N",
+#'   "PILOT01", "17-1344", "WEIGHT", "Weight (kg)", "WEEK 4", 57.97, "kg", "N",
+#'   "PILOT01", "17-1344", "WEIGHT", "Weight (kg)", "WEEK 6", 58.97, "kg", "N"
+#' )
+#'
+#' derive_vars_computed(
+#'   dataset = adsl,
+#'   dataset_add = advs,
+#'   by_vars = exprs(STUDYID, USUBJID),
+#'   parameters = c("WEIGHT"),
+#'   constant_by_vars = exprs(STUDYID, USUBJID),
+#'   constant_parameters = c("HEIGHT"),
+#'   new_vars = exprs(BMIBL = compute_bmi(height = AVAL.HEIGHT, weight = AVAL.WEIGHT)),
+#'   filter_add = ABLFL == "Y"
+#' )
+derive_vars_computed <- function(dataset,
+                                 dataset_add = NULL,
+                                 by_vars,
+                                 parameters,
+                                 new_vars,
+                                 filter_add = NULL,
+                                 constant_by_vars = NULL,
+                                 constant_parameters = NULL,
+                                 keep_nas = FALSE) {
```

Comment:
I would remove this argument as I am not aware of any use case where it makes sense to set it to `FALSE` (and it is hard to explain how it affects the results).